### PR TITLE
Focus hand detection

### PR DESCRIPTION
**Pull request by bluemarvin**
I'll rebase once #1149 lands.

### CODE REVIEW

**Comment by bluemarvin** [comment]
I think this is ready for review now.

**Comment by bluemarvin** [comment]
Actually, this isn't ready to land, it is broken on Oculus. Investigating.

**Comment by bluemarvin** [comment]
Fixed and tested on all platforms, googlevr, oculus, and wavevr.

File: app/src/wavevr/cpp/DeviceDelegateWaveVR.cpp [hunk 1]
```diff
@@ -368,21 +398,26 @@ DeviceDelegateWaveVR::ReleaseControllerDelegate() {
...
-    }
+    return "vr_controller_focus.obj";
+  } else if (aModelIndex == 1) {
     return "vr_controller_focus_plus.obj";
   }
-
   return "";
 }
 
+// #define VRB_WAVE_EVENT_LOG_ENABLED 1
```
Comment (daoshengmu) [inline app/src/wavevr/cpp/DeviceDelegateWaveVR.cpp]: nit: it seems like we don't need this comment.

Comment (bluemarvin) [reply app/src/wavevr/cpp/DeviceDelegateWaveVR.cpp]: We don't strictly need it but it makes it easier to turn the event logging on and off.

File: app/src/wavevr/cpp/DeviceDelegateWaveVR.cpp [hunk 2]
```diff
@@ -196,11 +220,33 @@ struct DeviceDelegateWaveVR::State {
       const bool touchpadTouched = WVR_GetInputTouchState(controller.type, WVR_InputId_Alias1_Touchpad);
       const bool menuPressed = WVR_GetInputButtonState(controller.type, WVR_InputId_Alias1_Menu);
 
-      delegate->SetButtonCount(index, 2); // For immersive mode
+
+      delegate->SetButtonCount(index, controller.is6DoF ? 3 : 2); // For immersive mode
```
[the remark this answers is not in the document]
Comment (bluemarvin) [reply app/src/wavevr/cpp/DeviceDelegateWaveVR.cpp]: The Focus Plus has and extra button on the grip.

[the remark this answers is not in the document]
Comment (cvan) [reply app/src/wavevr/cpp/DeviceDelegateWaveVR.cpp]: ah, thanks. a comment might be helpful. but it's good. as long as y'all know :smile:

**Pull request merged**